### PR TITLE
UI/UX improvements for the pipeline designer

### PR DESCRIPTION
**Pull request by flippingbits**
This PR introduces a couple of improvements to the UI/UX of our pipeline designer.

The major changes are:
* We show the `debug view` in the context bar instead of a modal:
Screenshot 2023-01-11 at 17 28 58
* We allow to delete pipeline steps without drag-and-drop, a feature that was very hard to discover for new users
* We move preview-related settings to the context bar: 
Screenshot 2023-01-11 at 17 28 44

### CODE REVIEW

File: ui/src/components/pipelines/pipeline_designer/context_bar/DebugView.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    if (this.state.tab === "error") {
+      debugContent = JSON.stringify(error, null, 2);
+    } else if (this.state.tab === "record") {
+      let originalRecord = JSON.parse(JSON.stringify(record));
+      delete originalRecord.metadata["error"];
+      delete originalRecord.id;
+      debugContent = JSON.stringify(originalRecord, null, 2);
+    } else if (this.state.tab === "transform") {
+      const match = /steps\[(\d+)\]\.fields\[(.+)\]/g.exec(
```
Comment (HknLof) [inline ui/src/components/pipelines/pipeline_designer/context_bar/DebugView.js:39]: Can you leave a comment, what this should match and what not?

File: ui/src/components/pipelines/pipeline_designer/context_bar/DebugView.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        error["location"]["path"]
+      );
+      if (match != null && match.length == 3) {
+        const step = parseInt(match[1]);
+        const fieldName = match[2];
+        debugContent = YAML.stringify(
+          pipeline.spec.steps[step].fields[fieldName]
+        );
+      } else {
+        const match = /steps\[(\d+)\].*/g.exec(error["location"]["path"]);
```
Comment (HknLof) [inline ui/src/components/pipelines/pipeline_designer/context_bar/DebugView.js:49]: Can you leave a comment, what this should match and what not?

**Review by HknLof — CHANGES_REQUESTED** [review]
Please describe the regex' and what these are supposed to catch and what they definitely should not match.

And these are warnings chrome console prints out in dev mode!
```
react-dom.development.js:88 Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move code with side effects to componentDidMount, and set initial state in the constructor.
* Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: BrowserRouter, Route, Router, Switch
printWarning @ react-dom.development.js:88
warn @ react-dom.development.js:51
__webpack_modules__../node_modules/react-dom/cjs/react-dom.development.js.ReactStrictModeWarnings.flushPendingUnsafeLifecycleWarnings @ react-dom.development.js:11371
flushRenderPhaseStrictModeWarningsInDEV @ react-dom.development.js:23112
commitRootImpl @ react-dom.development.js:22396
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
commitRoot @ react-dom.development.js:22381
finishSyncRender @ react-dom.development.js:21807
performSyncWorkOnRoot @ react-dom.development.js:21793
scheduleUpdateOnFiber @ react-dom.development.js:21188
updateContainer @ react-dom.development.js:24373
(anonymous) @ react-dom.development.js:24758
unbatchedUpdates @ react-dom.development.js:21903
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
./src/index.js @ index.js:12
options.factory @ react refresh:6
__webpack_require__ @ bootstrap:24
(anonymous) @ startup:7
(anonymous) @ startup:7
react-dom.development.js:88 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: Route, Router, Switch
```

File: ui/src/components/pipelines/pipeline_designer/context_bar/DebugView.js (hunk 1 above)
Comment (flippingbits) [reply ui/src/components/pipelines/pipeline_designer/context_bar/DebugView.js:44]: Thanks for raising this point. I hope the added comment is helpful?

**Review by HknLof — APPROVED** [review]

**Pull request merged**